gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/diagnosticreport/SpecimenCompoundsMapperTest.java

```diff
@@ -159,6 +169,53 @@ public void testHandlingNonUserNarrativeStatement() {
         assertThat(observation.getIssuedElement().asStringValue()).isNull();
     }
 
+    @Test public void testOrderingIsPreservedForDiagnosticReportResults() {
+        final RCMRMT030101UKEhrExtract  ehrExtract =
+            unmarshallEhrExtract("specimen_with_three_test_group_headers.xml");
+
+        final var testObservations = List.of(
+            (Observation) new Observation().setId("TEST-GROUP-HEADER-1"),
+            (Observation) new Observation().setId("OBSERVATION-STATEMENT-ID-1"),
+            (Observation) new Observation().setId("TEST-GROUP-HEADER-2"),
+            (Observation) new Observation().setId("OBSERVATION-STATEMENT-ID-2"),
+            (Observation) new Observation().setId("TEST-GROUP-HEADER-3"),
+            (Observation) new Observation().setId("OBSERVATION-STATEMENT-ID-3")
+        );
+
+        doAnswer(
+            answer(
+                (Answer1<Observation, SpecimenBatteryParameters>) batteryParameters -> {
+                    var id = batteryParameters.getBatteryCompoundStatement().getId().get(0).getRoot();
+                    var observation = new Observation();
+                    observation.setId(id);
+
+                    batteryParameters.getDiagnosticReport().addResult(
+                        new Reference(new IdType(ResourceType.Observation.name(), id)));
+                    return observation;
+                })
+        ).when(specimenBatteryMapper).mapBatteryObservation(any(SpecimenBatteryMapper.SpecimenBatteryParameters.class));
+
+        specimenCompoundsMapper.handleSpecimenChildComponents(
+            ehrExtract,
+            testObservations,
+            observationComments,
+            diagnosticReports,
+            PATIENT,
+            List.of(),
+            TEST_PRACTISE_CODE
+        );
+        var diagnosticReport = diagnosticReports.get(0);
```

Comment:
```suggestion
        );
        
        var diagnosticReport = diagnosticReports.get(0);
```

Reply:
I don't think we need a blank line here as that is still part of the "act stage" of AAA and so should be grouped together